src/main/java/pl/stormit/eduquiz/authenticator/controllers/AuthenticationViewController.java

```diff
@@ -0,0 +1,61 @@
+package pl.stormit.eduquiz.authenticator.controllers;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import pl.stormit.eduquiz.quizcreator.domain.user.UserService;
+import pl.stormit.eduquiz.quizcreator.domain.user.dto.UserRequestDto;
+
+@Controller
+@RequestMapping("/register")
+@AllArgsConstructor
+public class AuthenticationViewController {
+
+
+    private final UserService userService;
+
+    @GetMapping()
```

Comment:
this part of the code can be written like this:

    @GetMapping
    public String registerUser(final Model model, @ModelAttribute("error") String error) {
        UserRequestDto userRequestDto = new UserRequestDto(null, null, null, null, null, null, null);

        if (error != null && !error.isEmpty()) {
            model.addAttribute("error", error);
        }

        model.addAttribute("userRequestDto", userRequestDto);

        return "register";
    }
}

instead of the Error parameter (which is not a standard parameter in Spring MVC), we can use the @ModelAttribute annotation to pass errors as a model attribute; this way, you can pass it as an "error" parameter and later display some message in the view;